velox/exec/tests/FunctionSignatureBuilderTest.cpp

```diff
@@ -26,24 +27,20 @@ class FunctionSignatureBuilderTest : public functions::test::FunctionBaseTest {
 
 TEST_F(FunctionSignatureBuilderTest, basicTypeTests) {
   // All type variables should be used in the inputs arguments.
-  assertUserInvalidArgument(
-      [=]() {
-        FunctionSignatureBuilder()
-            .typeVariable("T")
-            .returnType("integer")
-            .argumentType("integer")
-            .build();
-      },
+  VELOX_ASSERT_THROW(
```

Comment:
don't we want to test the exception type being thrown anymore?

Reply:
VELOX_ASSERT_THROW ensures it is a VeloxException. Do you think we need to add VELOX_ASSERT_USER_THROW to ensure it is VeloxUserException?

```
#define VELOX_ASSERT_THROW(expression, errorMessage)                 \
  try {                                                              \
    (expression);                                                    \
    FAIL() << "Expected an exception";                               \
  } catch (const VeloxException& e) {                                \
    ASSERT_TRUE(e.message().find(errorMessage) != std::string::npos) \
        << "Expected error message to contain '" << errorMessage     \
        << "', but received '" << e.message() << "'.";               \
  }
```

Reply:
I think it's useful to check if we're throwing the right exception, so we don't mask errors where a different exception starts being thrown, but this might be fine since we're already checking the error message.

Reply:
@pedroerp How about we merge this change and do a follow-up to introduce VELOX_ASSERT_USER_THROW?